From 575a5bb38ad82ae636131c7330aed73ded8e0e4b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 21 Dec 2023 10:59:18 +0100
Subject: [PATCH] update-kernel: support more System.map and config-* files

Also support Sytem.map and config with kernel release suffix.
---
 tests/bin/apk    | 2 +-
 update-kernel.in | 9 ++++-----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/tests/bin/apk b/tests/bin/apk
index 9868230..aaed957 100755
--- a/tests/bin/apk
+++ b/tests/bin/apk
@@ -85,7 +85,7 @@ for pkg in $pkgs; do
 			flavor=${pkg#linux-}
 			mkdir -p "$rootfs"/lib/modules/$kver-$flavor/kernel/drivers/net/wireless/broadcom/brcm80211/brcmfmac \
 				"$rootfs"/boot
-			touch "$rootfs"/boot/System.map-$flavor \
+			touch "$rootfs"/boot/System.map-$kver-$flavor \
 				"$rootfs"/boot/config-$flavor \
 				"$rootfs"/boot/vmlinuz-$flavor
 			cat >"$rootfs/lib/modules/$kver-$flavor"/kernel/drivers/net/wireless/broadcom/brcm80211/brcmfmac/brcmfmac.ko <<-EOF
diff --git a/update-kernel.in b/update-kernel.in
index ae4ed0f..cb300a1 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -351,11 +351,10 @@ $MOCK mkinitfs $MKINITFS_ARGS -q -b $ROOTFS -F "$features base squashfs" \
 	-o "$STAGING/initramfs-$FLAVOR" "$KVER"
 
 for file in System.map config vmlinuz; do
-	if [ -f "$BOOT/$file-$FLAVOR" ]; then
-		cp "$BOOT/$file-$FLAVOR" $STAGING
-	else
-		cp "$BOOT/$file" $STAGING
-	fi
+	cp "$BOOT/$file-$KVER" "$STAGING" 2>/dev/null \
+		|| cp "$BOOT/$file-$FLAVOR" "$STAGING" 2>/dev/null \
+		|| cp "$BOOT/$file" "$STAGING" 2>/dev/null \
+		|| { echo "Failed to copy $file-$KVER, $file-$FLAVOR or $file" >&2; exit 1; }
 done
 
 if [ "$MNTDIR" ]; then
-- 
2.43.0

